presto-password-authenticators/src/test/java/io/prestosql/plugin/password/TestFileAuthConfig.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.airlift.log.Logger;
+import io.prestosql.plugin.password.filebasedauthenticator.EncryptionHelper;
+import io.prestosql.plugin.password.filebasedauthenticator.FileBasedAuthenticationConfig;
+import io.prestosql.plugin.password.filebasedauthenticator.HashedPasswordException;
+import io.prestosql.plugin.password.filebasedauthenticator.User;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+import static io.prestosql.plugin.password.filebasedauthenticator.HashingAlgo.BCRYPT;
+import static io.prestosql.plugin.password.filebasedauthenticator.HashingAlgo.PBKDF2;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestFileAuthConfig
+{
+    private static final Logger log = Logger.get(TestFileAuthConfig.class);
+    private long maximumSize = 1000;
+    public Cache<User.PasswordStore, Boolean> bcryptCache;
+    public Cache<User.PasswordStore, Boolean> pbkdf2Cache;
+    public int bcryptMinCost = 8;
+    public int pbkdf2MinIterations = 1000;
+    FileBasedAuthenticationConfig config;
```

Comment:
why some fields are public, some are private and some are package-private?

Reply:
All private now